openedx/features/qverse_features/registration/validators.py

```diff
@@ -1,34 +1,64 @@
 """
 Validation utils for qverse registration application.
 """
+from csv import reader
+import io
+
 from django.core.exceptions import ValidationError
 
 
-def validate_file_extension(file):
+def validate_admission_file(file):
     """
-    Validates that file is a valid csv file by checking its extension.
+    Validates that the file is a valid csv file by checking its extension.
+
+    Also checks that the header row has all the required fields.
+
+    Arguments:
+        file (models.FileField): The file uploaded by the user
     """
     if not file.name.endswith('.csv'):
         raise ValidationError('', code='invalid')
 
+    FIELD_NAMES = [
+                'regno', 'firstname', 'surname', 'othername', 'levelid',
+                'programmeid', 'departmentid', 'mobile', 'email', 'mobile'
+                ]
+    decoded_file = file.read().decode('utf-8')
+    io_string = io.StringIO(decoded_file)
```

Comment:
why we need decoding here?

Reply:
Not sure about that but found this code at two different places and both followed the same approach.

Reply:
done.

Reply:
`Seek first to understand, then to be understood` :)